remote/mattermost/remote.go

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package mattermost
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/target/flottbot/models"
+	"github.com/target/flottbot/remote"
+)
+
+// Client struct.
+type Client struct {
+	Server   string
+	Token    string
+	BotID    string
+	Insecure bool
+}
+
+type mmLogger struct {
+	zerolog.Logger
+}
+
+func (l *mmLogger) Output(_ int, s string) error {
+	l.Logger.Info().Msg(s)
+	return nil
+}
+
+// validate that Client adheres to remote interface.
+var _ remote.Remote = (*Client)(nil)
+
+// instantiate a new mattermost client.
+func (c *Client) new() *model.Client4 {
+	log.Info().Msgf("%#v", c)
+
+	var url string
+	if c.Insecure {
+		url = "http://" + c.Server
+	} else {
+		url = "https://" + c.Server
+	}
+
+	log.Info().Msgf("connecting to instance with url: %s", url)
+	api := model.NewAPIv4Client(url)
+	api.SetToken(c.Token)
+
+	return api
+}
+
+func (c *Client) Name() string { return "mattermost" }
+
+func (c *Client) Reaction(_ models.Message, rule models.Rule, _ *models.Bot) {
+	if rule.RemoveReaction != "" {
+		log.Debug().Msg("Remove reaction not implemented for mattermost")
+	}
+
+	if rule.Reaction != "" {
+		log.Debug().Msg("Reactions not implemented for mattermost")
+	}
+
+}
+
+func (c *Client) Read(inputMsgs chan<- models.Message, _ map[string]models.Rule, bot *models.Bot) {
+	api := c.new()
+	if user, resp, err := api.GetUser("me", ""); err != nil {
+		log.Fatal().Msgf("Could not login, %s", err)
+	} else {
+		log.Info().Interface("user", user.Username).Interface("resp", resp).Msg("")
+		log.Info().Msg("Logged in to mattermost")
+
+		c.BotID = user.Username
+	}
```

Comment:
```suggestion
	if user, resp, err := api.GetUser("me", ""); err != nil {
		log.Fatal().Msgf("could not login, %s", err)
	}
	
	log.Info().Interface("user", user.Username).Interface("resp", resp).Msg("")
	log.Info().Msg("logged in to mattermost")

	c.BotID = user.Username
```

i think we can lose the `else` here